src/io/flutter/run/FlutterRunConfigurationProducer.java

```diff
@@ -51,6 +52,12 @@ public boolean isConfigurationFromContext(final @NotNull FlutterRunConfiguration
     return dartFile != null && dartFile.getPath().equals(configuration.getRunnerParameters().getFilePath());
   }
 
+  @Override
+  public boolean shouldReplace(ConfigurationFromContext self, ConfigurationFromContext other) {
+    // Prefer Flutter runner to plain Dart runner for Flutter code.
+    return other.getConfigurationType().getId().equals("DartCommandLineRunConfigurationType");
```

Comment:
Looking at this more closely, it seems this class already uses a lot of other Dart plugin code? For example, DartFile, DartImportStatement, DartResolveUtil.